src/ng/directive/validators.js

```diff
@@ -353,21 +387,49 @@ var maxlengthDirective = function() {
  *   </file>
  * </example>
  */
-var minlengthDirective = function() {
+var minlengthDirective = function($parse) {
   return {
     restrict: 'A',
     require: '?ngModel',
     link: function(scope, elm, attr, ctrl) {
       if (!ctrl) return;
 
-      var minlength = 0;
+      var minlength = attr.minlength || $parse(attr.ngMinlength)(scope);
+      var minlengthParsed = toInt(minlength) || 0;
+
       attr.$observe('minlength', function(value) {
-        minlength = toInt(value) || 0;
-        ctrl.$validate();
+        if (minlength !== value) {
+          minlengthParsed = toInt(value) || 0;
+          minlength = value;
+          ctrl.$validate();
+        }
+
       });
       ctrl.$validators.minlength = function(modelValue, viewValue) {
-        return ctrl.$isEmpty(viewValue) || viewValue.length >= minlength;
+        return ctrl.$isEmpty(viewValue) || viewValue.length >= minlengthParsed;
       };
     }
   };
 };
+
+
+function parsePatternAttr(regex, patternExp, elm) {
+  if (!regex) return undefined;
+
+  if (isString(regex) && regex.length > 0) {
+    regex = new RegExp('^' + regex + '$');
+  }
+
+  if (!(regex instanceof RegExp)) {
```

Comment:
Checking for `instanceof RegExp` instead of `regep.test` looks like a breaking change (and an unnecessary one afaict :grin:).

Reply:
We only allow RegExp and regex literals, which both are identified correctly by instanceof. Or could there be another reason for having .test instead?

Reply:
I mean someone might be taking advantage of the fact that we were only testing for `.test` and provide a custom object like `{test: () => /* Do some fancy stuff */}`.
I know it is not a great idea, but I also know I've done it in the past :grin:

What I am saying is that I don't see any benefit is breaking this (admittedly dubious) usecase at this point (a few months into LTS already).
Is there any benefit in changing that?